fdbserver/BulkDumpUtil.actor.cpp

```diff
@@ -0,0 +1,229 @@
+                                              int64_t dumpBytes) {
+	const std::string dumpFolder = abspath(joinPath(rootFolder, relativeFolder));
+	platform::eraseDirectoryRecursive(dumpFolder);
+	if (!platform::createDirectory(dumpFolder)) {
+		throw retry();
+	}
+
+	std::string dataFilePath = "";
+	std::string byteSampleFilePath = "";
+	if (sortedKVS.size() > 0) {
+		std::string dataFileName = generateBulkDumpDataFileName(dumpVersion);
+		dataFilePath = abspath(joinPath(dumpFolder, dataFileName));
+		ASSERT(!fileExists(dataFilePath));
+		std::string byteSampleFileName = generateBulkDumpByteSampleFileName(dumpVersion);
+		byteSampleFilePath = abspath(joinPath(dumpFolder, byteSampleFileName));
+		ASSERT(!fileExists(byteSampleFilePath));
+
+		std::unique_ptr<IRocksDBSstFileWriter> sstWriterData = newRocksDBSstFileWriter();
+		std::unique_ptr<IRocksDBSstFileWriter> sstWriterByteSample = newRocksDBSstFileWriter();
+		sstWriterData->open(dataFilePath);
+		sstWriterByteSample->open(byteSampleFilePath);
+		bool anySampled = false;
+		for (const auto& [key, value] : sortedKVS) {
+			sstWriterData->write(key, value); // assuming sorted
+			ByteSampleInfo sampleInfo = isKeyValueInSample(KeyValueRef(key, value));
+			if (sampleInfo.inSample) {
+				sstWriterByteSample->write(key, value);
+				anySampled = true;
+			}
+		}
+		ASSERT(sstWriterData->finish());
+		if (anySampled) {
+			ASSERT(sstWriterByteSample->finish());
+		} else {
+			ASSERT(deleteFile(byteSampleFilePath));
+		}
+	}
+
+	std::string manifestFileName = generateBulkDumpManifestFileName(dumpVersion);
+	std::string manifestFilePath = abspath(joinPath(dumpFolder, manifestFileName));
+	ASSERT(!fileExists(manifestFilePath));
+
+	BulkDumpManifest manifest(
+	    dataFilePath, manifestFilePath, byteSampleFilePath, dumpRange.begin, dumpRange.end, dumpVersion, "", dumpBytes);
+	writeFile(manifestFilePath, manifest.toString());
+
+	return manifest;
+}
+
+void bulkDumpFileCopy(std::string fromFile, std::string toFile, size_t fileBytesMax) {
+	std::string content = readFileBytes(fromFile, fileBytesMax);
+	writeFile(toFile, content);
+	return;
+}
+
+ACTOR Future<Void> bulkDumpTransportCP_impl(std::string fromRoot,
+                                            std::string toRoot,
+                                            std::string relativeFolder,
+                                            size_t fileBytesMax,
+                                            UID logId) {
+	loop {
+		state std::string toFolder = abspath(joinPath(toRoot, relativeFolder));
+		state std::string fromFolder = abspath(joinPath(fromRoot, relativeFolder));
+		state std::string fromFile;
+		state std::string toFile;
+		try {
+			// Clear existing folder
+			platform::eraseDirectoryRecursive(toFolder);
+			if (!platform::createDirectory(toFolder)) {
+				throw retry();
+			}
+
+			// Move bulk dump files to the target folder
+			for (const auto& filePath : platform::listFiles(fromFolder)) {
```

Comment:
nit: we don't use std::filesystem::cp for this kind of thing?

Reply:
I feel platform::listFiles is standard in FDB implementation? I am thinking..